pkg/secrets-store/nodeserver.go

```diff
@@ -259,14 +264,30 @@ func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 	targetPath := req.GetTargetPath()
 	volumeID := req.GetVolumeId()
 
+	// check if target path is mount point
+	notMnt, err := mount.New("").IsLikelyNotMountPoint(targetPath)
+	if (err != nil && os.IsNotExist(err)) || notMnt {
+		log.Infof("target path %s is not likely a mount point, notMnt %v", targetPath, notMnt)
+		return &csi.NodeUnpublishVolumeResponse{}, nil
+	}
+	if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+	// if mount point, ensure mount link is still valid
+	if !notMnt {
+		if _, err = ioutil.ReadDir(targetPath); err != nil {
```

Comment:
If the target is mounted, when would it fail on read?

Reply:
As long it's a valid mnt point, we should unmount it. So removed this check.